### PR TITLE
Minor cleanup in the monitoring PML.

### PR DESCRIPTION
**Pull request by bosilca**
Signed-off-by: George Bosilca <bosilca@icl.utk.edu>

### CODE REVIEW

File: ompi/mca/common/monitoring/common_monitoring.c [hunk 1]
```diff
@@ -253,7 +253,7 @@ void mca_common_monitoring_finalize( void )
     }
 }
 
-void mca_common_monitoring_register(void*pml_monitoring_component)
+int mca_pml_monitoring_register(void)
```
Comment (gpaulsen) [inline ompi/mca/common/monitoring/common_monitoring.c]: Seems strange to define a `mca_pml_monitoring_register()` function in the `common_monitoring.c`, but the common files are kinda a wild west, so go for it.

Comment (bosilca) [reply ompi/mca/common/monitoring/common_monitoring.c]: I could indeed change the name to reflect the real meaning of this function. One problem remains however, all these MCA parameters are tied to the monitoring PML, as the PML is required for the entire monitoring infrastructure to work.

Comment (gpaulsen) [reply ompi/mca/common/monitoring/common_monitoring.c]: It's fine by me as is.

Comment (bosilca) [reply ompi/mca/common/monitoring/common_monitoring.c]: Really funny thing we got here. I tried to fix this, and move the MCA variables out of the PML scope into the common/monitoring scope. Well, that didn't work well, the common scope is removed last, after the last usage of any user of the common support, which means a moment where the common shared libraries are already dlclosed. And this, generate segfaults on all string registrations.

Comment (jsquyres) [reply ompi/mca/common/monitoring/common_monitoring.c]: @hjelmn ...?  😄

**Pull request merged**